enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,449 @@
+
+### Goals  
+
+1. Enable administrators to configure and validate advanced authentication settings, including `DiscoveryURL`, `AudienceMatchPolicy`, `UID`, `Extra` claims, `ClaimValidationRules`, and `UserValidationRules`.  
+2. Support flexible claim mapping, multi-cluster authentication, and integration with external identity providers.  
+3. Strengthen security by enforcing advanced validation rules and identity policies to ensure proper access control.  
+
+### Non-Goals  
+
+- This enhancement does **not** introduce new authentication mechanisms beyond OIDC.  
+
+## Proposal
+
+This proposal introduces missing authentication fields to the OpenShift API by modifying the `authentications.config.openshift.io` CustomResourceDefinition (CRD) and updating relevant components. These changes will improve support for advanced OIDC configurations, identity customization, and security enforcement.
+
+### Changes to `authentications.config.openshift.io` CRD  
+
+To enhance authentication flexibility and security, the following fields will be added to the CRD:  
+
+- **Issuer Configuration**  
+  - `.spec.oidcProviders[].issuer.discoveryURL`: Allows specifying a custom OIDC discovery endpoint.  
+  - `.spec.oidcProviders[].issuer.audienceMatchPolicy`: Enables flexible audience validation rules.  
+
+- **Claim and User Validation Rules**  
+  - `.spec.oidcProviders[].claimValidationRules.expression`: Supports CEL-based validation of claims.  
+  - `.spec.oidcProviders[].userValidationRules`: Enforces security policies on usernames and groups.  
+
+### Components to Be Updated  
+
+To ensure the new authentication fields are processed and applied, the following components need modifications:  
+
+1. **Cluster Authentication Operator**  
+   - Today, the Cluster Authentication Operator is responsible for generating the authentication configuration by pulling settings from the `authentication.config.openshift.io` CRD and writing the corresponding configuration files used by the `kube-apiserver`.  
+   - This logic will be updated to ensure that the new fields (`discoveryURL`, `audienceMatchPolicy`, claim mappings, and validation rules) are correctly extracted from the CRD and included in the generated authentication configuration passed to the `kube-apiserver`.  
+
+2. **Hypershift Control Plane Operator**  
+   - Today, the Hypershift Control Plane Operator is responsible for generating and managing authentication configurations for hosted control planes, ensuring consistency across managed clusters.  
+   - This logic will be updated to support the new fields (`discoveryURL`, `audienceMatchPolicy`, claim mappings, and validation rules) so that authentication policies remain aligned across hosted control planes. The operator will extract these configurations from the `authentication.config.openshift.io` CRD and ensure they are correctly propagated to the hosted control plane's authentication setup.  
```

Comment:
```suggestion
   - This logic will be updated to support the new fields (`discoveryURL`, `audienceMatchPolicy`, and validation rules) so that authentication policies remain aligned across hosted control planes. The operator will extract these configurations from the `authentication.config.openshift.io` CRD and ensure they are correctly propagated to the hosted control plane's authentication setup.  
```